lib/batch/add_publisher_for_collection_members.rb

```diff
@@ -0,0 +1,18 @@
+# updates all the items in a collection with no publisher value.
+# Requires a collection_id
+
+def add_publisher_for_collection_members(collection_id:)
+  collection = Collection.find(collection_id)
+
+  puts "Checking #{collection.members.count} items in #{collection.title} for empty publisher"
+  collection.members.each do |member|
+    if member.class == Collection
+      add_publisher_for_collection_members(collection_id: member.id)
+    end
+
+    if member.publisher.empty?
+      member.publisher += ["DigitalHub. Galter Health Sciences Library & Learning Center"]
+      member.save!
```

Comment:
Two comments about this, but I don't feel strongly about either. Just things to consider.

I'd recommend using `#blank?` and not `#empty?` here. The difference of these is that empty will only return true if it's a completely empty string or nil, but blank will also return true if it's all white space characters. This will probably not matter here, but it's something to consider.

It's not really a big difference, but you can save a line by doing `member.update!(publisher: "...")`. You could save even more lines with a post conditional: `member.update(publisher: "") if member.publisher.blank?`.

Reply:
Ah, I do not think I ever knew the difference with `#blank?` and `#empty?`. I like the update suggestion, so I will make the change.